reana_workflow_engine_snakemake/executor.py

```diff
@@ -216,8 +219,8 @@ def _generate_report(workflow_file_path):
         cluster="reana",
         config=workflow_parameters,
         workdir=workflow_workspace,
-        immediate_submit=True,
         notemp=True,
+        nodes=4,  # enables DAG parallelization
```

Comment:
I'm not very sure of the number of `nodes` here. I chose 4 in the same line [we chose 4 cores when running the example locally](https://github.com/reanahub/reana-demo-cms-h4l/blob/b6ed43426c0fda4e5e144e34a85e90e21666ba61/workflow/snakemake/Snakefile#L11).

Reply:
Shouldn't we specify the number of our cluster worker nodes here? I did a quick test with your dummy scatter-gather example which should spawn 5 jobs in parallel, but only 4 was spawned because of this setting. When I increased it to 6, all 5 jobs were running in parallel. So it could be a big constraint in the future for large workflows. We could perhaps create a separate issue for that? Since the goal to enable parallelization was achieved nicely here!

Reply:
I guess it needs deeper discussions, yes, I'll create a ticket.